cli/azd/pkg/project/service.go

```diff
@@ -88,7 +89,13 @@ func (svc *Service) Deploy(ctx context.Context, azdCtx *environment.AzdContext)
 // If not found will assume resource name conventions
 func GetServiceResourceName(ctx context.Context, resourceGroupName string, serviceName string, env *environment.Environment) (string, error) {
 	azCli := commands.GetAzCliFromContext(ctx)
-	query := fmt.Sprintf("resources | where resourceGroup == '%s' | where tags['azd-service-name'] == '%s' | project id, name, type, tags, location", resourceGroupName, serviceName)
+	query := fmt.Sprintf(
+		"resources | where resourceGroup == '%s' | where tags['azd-service-name'] == '%s' | "+
+			"project id, name, type, tags, location",
+		// lowercase resource name as using capitals would result in resource group not found
+		// see: https://github.com/Azure/azure-dev/issues/115
+		strings.ToLower(resourceGroupName),
```

Comment:
Might be good to update the comment on why there's a casing mismatch. I believe the problem is that `azd` is handling casing correctly, and the resource groups are created with the correct casing, but simply that this graph query command we're using only produces lower case letters? That might be a better explanation.

Reply:
This does seem to be the case. The Resource Graph queries have resource groups all lower-cased:
```
id               : /subscriptions/faa080af-c1d8-40ad-9cce-e1a450ca5b57/resourceGroups/WeiLim-Capital-Resources-rg/providers/microsoft.alertsmanagement/smart
                   DetectorAlertRules/Failure Anomalies - appi-2ufe2mjspqjtw
name             : Failure Anomalies - appi-2ufe2mjspqjtw
type             : microsoft.alertsmanagement/smartdetectoralertrules
tenantId         : 72f988bf-86f1-41af-91ab-2d7cd011db47
kind             :
location         : global
resourceGroup    : weilim-capital-resources-rg
subscriptionId   : faa080af-c1d8-40ad-9cce-e1a450ca5b57
managedBy        :
sku              :
plan             :
properties       : @{description=Failure Anomalies notifies you of an unusual rise in the rate of failed HTTP requests or dependency calls.; state=Enabled;
                   actionGroups=; severity=Sev3; frequency=PT1M; throttling=; detector=; scope=System.Object[]}
tags             :
identity         :
zones            :
extendedLocation :
```

Reply:
@ellismg do you have context of why we're using Azure Resource Grap here? It seems that Azure Resource Graph is primarily used for governance purposes, the query we're performing here can simply be substituted with an `az resource` command: `az resource list --resourceGroup <rg> --tags <tag>`. 

The bug here that we've uncovered also seems to be a bug on the service (I couldn't find any documentation detailing lowercased resource group names being the design), and could not find a place to file an issue (other than going through internal channels). Seems like a good idea to use the more supported `az resource` API.

Reply:
@weikanglim we need to somehow provide feedback for the Resource Graph about this.  The resource graph is used as the Azure Portal Search engine, so it is not only Governance purposes.

IMO, it gives a general solution, and we should keep using it.

I found [this documentation](https://docs.microsoft.com/en-us/rest/api/azureresourcegraph/resourcegraph(2021-03-01)/resources/resources?tabs=HTTP#query-with-a-facet-request), note how there's a `tolower()` for resource group:
![image](https://user-images.githubusercontent.com/24213737/179566912-c8a365c4-3c16-4fe9-ad8e-f6dd31a5c593.png)

This means that it might be as per design.  But I need to find more info...  Probably @jongio knows people from Resource Graph...

Reply:
We are using the resource graph here because the resource API does not support both querying by resource group and tags.  `resource` api requires you get a list of resources within an RG or to query all resources by tag.

Reply:
@weikanglim - @wbreza added the code initially, so he would know exactly what he was thinking, but I probably would have done the same thing. I guess my rationale here is that us moving to this tagging model makes it so these sorts of queries could be performed in a uniform way across all resource types and that sort of technology would be very powerful to have in the future.  I could also imagine that the graph query could be more preformat since filtering could be done server rather than client side (I assume that `az resource list` does the tag filtering locally, and you could imagine the graph API having some better state management system that makes these queries faster).  But I think that moving to `az resource list` may make sense here.

Wallace, do you think there any gotchas here?

Reply:
@vhvb1989 I found the internal docs portal, the contact is listed on [here](https://eng.ms/docs/cloud-ai-platform/azure-core/azure-management-and-platforms/control-plane-bburns/azure-resource-graph/azure-resource-graph/arn-wiki/contact-us ).

@wbreza @ellismg Thanks for all the comments. Given the extensibility in querying and more performant nature of Azure Resource Graph, I think we should keep the dependency, follow up with the Azure Resource Graph team on the resource groups usage, and update the current code comment to reflect the "gotcha" encountered.